[BUG] test: flaky test in Test_multiplyConditionRoute_route/test_multiply_destination_with_ignore_some_condition

### ✅ 验证清单

- [x] 🔍 我已经搜索过 [现有 Issues](https://github.com/apache/dubbo-go/issues)，确信这不是重复问题

### 🚀 Go 版本

1.24+

### 📦 Dubbo-go 版本

3.3.1

### 🖥️ 服务端配置

_No response_

### 💻 客户端配置

_No response_

### 🌐 协议配置

_No response_

### 📋 注册中心配置

_No response_

### 💾 操作系统

🪟 Windows

### 📝 Bug 描述

 ## What happened
The test `Test_multiplyConditionRoute_route/test_multiply_destination_with_ignore_some_condition` in `cluster/router/condition/router_test.go` is flaky and fails intermittently in CI.
https://github.com/apache/dubbo-go/actions/runs/21290486533/job/61282574758?pr=3175
##  Error message
--- FAIL: Test_multiplyConditionRoute_route (0.57s)
    --- FAIL: Test_multiplyConditionRoute_route/test_multiply_destination_with_ignore_some_condition (0.21s)
        router_test.go:1374: 
            Error Trace:    cluster/router/condition/router_test.go:1374
            Error:          out of range


### Related log:
ERROR condition/dynamic_router.go:86 requestconsumer://:/com.foo.BarService?env=gray&region=beijing 
      route an empty set in condition-route:: request=env=gray,invokers!=env=ErrTag

### 🔄 重现步骤

CI environment: GitHub Actions

This flaky test was observed in PR #3175, which only modifies filter/generic/ and is completely unrelated to cluster/router/condition/.

### ✅ 预期行为

ci 通过

### ❌ 实际行为

 ci 没通过

### 💡 可能的解决方案

_No response_

--- a/cluster/router/condition/router_test.go
+++ b/cluster/router/condition/router_test.go
@@ -1350,17 +1350,21 @@ conditions:
 				}
 			} else {
 				// check multiply destination route successfully or not
-				ans := map[any]float32{}
+				// Use larger sample size for statistical stability
+				const sampleSize = 5000
+				ans := map[int]float32{}
 				for _, s := range tt.multiDestination {
 					args := struct {
 						invokers   []base.Invoker
 						url        *common.URL
 						invocation base.Invocation
 					}{tt.args.invokers[:], tt.args.url.Clone(), tt.args.invocation}
-					ans[len(s.invokers_filters.filtrate(args.invokers, tt.args.url, tt.args.invocation))] = s.weight * 1000
+					key := len(s.invokers_filters.filtrate(args.invokers, tt.args.url, tt.args.invocation))
+					// Accumulate weights for same-length results to avoid map key collision
+					ans[key] += s.weight * sampleSize
 				}
-				res := map[any]int{}
-				for i := 0; i < 1000; i++ {
+				res := map[int]int{}
+				for i := 0; i < sampleSize; i++ {
 					args := struct {
 						invokers   []base.Invoker
 						url        *common.URL
@@ -1368,11 +1372,17 @@ conditions:
 					}{tt.args.invokers[:], tt.args.url.Clone(), tt.args.invocation}
 					res[len(d.Route(args.invokers, args.url, args.invocation))]++
 				}
-				for k, v := range ans {
-					if float32(res[k]+50) > v && float32(res[k]-50) < v {
-					} else {
-						assert.Fail(t, "out of range")
+				for k, expected := range ans {
+					actual := float32(res[k])
+					// Use percentage-based tolerance: max(25% of expected, 50) to handle both
+					// high and low expected values while accounting for statistical variance
+					tolerance := expected * 0.25
+					if tolerance < 50 {
+						tolerance = 50
 					}
+					assert.Truef(t, actual >= expected-tolerance && actual <= expected+tolerance,
+						"result count out of range for key=%d: expected=%.0f, actual=%.0f, tolerance=%.0f",
+						k, expected, actual, tolerance)
 				}
 			}
 		})

